### PR TITLE
[SPARK-40305][PS] Implement Groupby.sem

### PR DESCRIPTION
**Pull request by zhengruifeng**
### What changes were proposed in this pull request?
Implement Groupby.sem

### Why are the changes needed?
to increase API coverage

### Does this PR introduce _any_ user-facing change?
yes, new API

### How was this patch tested?
added UT

### CODE REVIEW

**Comment by zhengruifeng** [comment]
cc @itholic @xinrong-meng @HyukjinKwon

File: python/pyspark/pandas/groupby.py [hunk 1]
```diff
@@ -827,6 +827,76 @@ def mad(self) -> FrameLike:
...
+        if ddof not in [0, 1]:
+            raise TypeError("ddof must be 0 or 1")
+
+        # Raise the TypeError when all aggregation columns are of unaccepted data types
+        all_unaccepted = True
+        for _agg_col in self._agg_columns:
+            if isinstance(_agg_col.spark.data_type, (NumericType, BooleanType)):
+                all_unaccepted = False
+                break
+        if all_unaccepted:
```
Comment (xinrong-meng) [inline python/pyspark/pandas/groupby.py:879]: nit: 
```
any_accepted = any(isinstance(_agg_col.spark.data_type, (NumericType, BooleanType)) for _agg_col in self._agg_columns)
if not any_accepted:
  raise TypeError("...")
```
?
Feel free to keep the existing approach if you prefer :)

File: python/pyspark/pandas/generic.py [hunk 2]
```diff
@@ -2189,7 +2189,7 @@ def std(psser: "Series") -> Column:
                 return F.stddev_samp(spark_column)
 
         def sem(psser: "Series") -> Column:
-            return std(psser) / pow(Frame._count_expr(psser), 0.5)
+            return std(psser) / F.sqrt(Frame._count_expr(psser))
```
Comment (xinrong-meng) [inline python/pyspark/pandas/generic.py:2192]: Nice!

File: python/pyspark/pandas/tests/test_groupby.py [hunk 3]
```diff
@@ -3055,6 +3065,17 @@ def test_ddof(self):
                 psdf.groupby("a")["b"].var(ddof=ddof).sort_index(),
                 check_exact=False,
             )
+            # var
```
Comment (xinrong-meng) [inline python/pyspark/pandas/tests/test_groupby.py:3068]: nit: `# sem`

File: python/pyspark/pandas/groupby.py (hunk 1 above)
Comment (zhengruifeng) [reply python/pyspark/pandas/groupby.py:879]: nice, this is more concise. I just copied it from other places

**Review by xinrong-meng — APPROVED** [review]

**Comment by zhengruifeng** [comment]
Merged into master, thank you @xinrong-meng for reivews

**Comment by itholic** [comment]
Just switched this issues (SPARK-40305) from "improvement" to "sub-task" of [Increase pandas API coverage for pandas API on Spark](https://issues.apache.org/jira/browse/SPARK-40327) in the ticket.

Please feel free to revert it if it's not a correct place.

**Comment by zhengruifeng** [comment]
> Just switched this issues ([SPARK-40305](https://issues.apache.org/jira/browse/SPARK-40305)) from "improvement" to "sub-task" of [Increase pandas API coverage for pandas API on Spark](https://issues.apache.org/jira/browse/SPARK-40327).
> 
> Please feel free to revert if it's not a correct switching!

Sure, that is fine!

**Pull request closed without merging**